### PR TITLE
Add the F distribution functions

### PR DESCRIPTION
**Pull request by talgalili**
Co-authored-by: Masha Iris <Masha-Iris@users.noreply.github.com>

Based on the diff: #15958

Test plan: new tests are included in the diff.

For importance, see: https://en.wikipedia.org/wiki/F_distribution.

This diff aims to resolve this (2 years old) issue: #15959. (this is a fixed version of this diff: #15958)

```
== RELEASE NOTES ==

General Changes
* Add f_cdf() and inverse_f_cdf() functions.
```

### CODE REVIEW

**Comment by talgalili** [comment]
Numbers used in the tests were taken from running the following R code:

```r
> 
> round(qf(0.5, 2,5),4)
[1] 0.7988
> round(qf(0.5, 2,5), 4)
[1] 0.7988
> 
> round(qf(0, 2,5), 4)
[1] 0
> 
> round(qf(0.9, 2,5), 4)
[1] 3.7797
> round(pf(0.7988, 2,5), 4)
[1] 0.5
> round(pf(3.7797, 2,5), 4)
[1] 0.9

```

**Comment by talgalili** [comment]
Errors in:
https://github.com/prestodb/presto/actions/runs/4441048435/jobs/7795625655?pr=19203
Seems related to style checks, in files other than the ones I've changed.

**Comment by rschlussel** [comment]
These are the checkstyle errors from the logs

```
[INFO] There are 5 errors reported by Checkstyle 8.16 with src/checkstyle/presto-checks.xml ruleset.
Error:  src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:[38] (imports) ImportOrder: Wrong order for 'org.apache.commons.math3.distribution.FDistribution' import.
Error:  src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:[910,51] (naming) ParameterName: Name 'n_df' must match pattern '^[a-z][a-zA-Z0-9]*$'.
Error:  src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:[911,51] (naming) ParameterName: Name 'd_df' must match pattern '^[a-z][a-zA-Z0-9]*$'.
Error:  src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:[925,51] (naming) ParameterName: Name 'n_df' must match pattern '^[a-z][a-zA-Z0-9]*$'.
Error:  src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:[926,51] (naming) ParameterName: Name 'd_df' must match pattern '^[a-z][a-zA-Z0-9]*$'.
```

**Comment by talgalili** [comment]
Thanks @rschlussel !

Sorry I've missed those.
I attempted to fix, and push again. Let's see if it worked.

p.s.: I also found some typos in the docs, and proposed a fix for them here - 
https://github.com/prestodb/presto/pull/19205

**Comment by talgalili** [comment]
Hey @feilong-liu - could you please take a look (and hopefully accept) this diff?

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java [hunk 1]
```diff
@@ -902,6 +903,36 @@ public static double chiSquaredCdf(
         return distribution.cumulativeProbability(value);
     }
 
+    @Description("Inverse of F cdf given numerator df, denominator df parameters, and probability")
```
Comment (feilong-liu) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:906]: numerator df, denominator df -> numerator df**1**, denominator df**2**

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java [hunk 2]
```diff
@@ -902,6 +903,36 @@ public static double chiSquaredCdf(
...
+            @SqlType(StandardTypes.DOUBLE) double p)
+    {
+        checkCondition(p >= 0 && p <= 1, INVALID_FUNCTION_ARGUMENT, "p must be in the interval [0, 1]");
+        checkCondition(df1 > 0, INVALID_FUNCTION_ARGUMENT, "numerator df must be greater than 0");
+        checkCondition(df2 > 0, INVALID_FUNCTION_ARGUMENT, "denominator df must be greater than 0");
+        FDistribution distribution = new FDistribution(null, df1, df2, FDistribution.DEFAULT_INVERSE_ABSOLUTE_ACCURACY);
+        return distribution.inverseCumulativeProbability(p);
+    }
+
+    @Description("F cdf given the numerator df, denominator df parameters, and value")
```
Comment (feilong-liu) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:921]: same as above

**Review by feilong-liu — APPROVED** [review]
Some nit, otherwise lgtm

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java (hunk 1 above)
Comment (talgalili) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java]: Thanks, fixed.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java (hunk 2 above)
Comment (talgalili) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java]: Thanks, fixed.

**Comment by talgalili** [comment]
Thanks for the review @feilong-liu 

@rschlussel - I believe the diff is now ready for landing. Thanks!

**Comment by talgalili** [comment]
Thanks @rschlussel

**Pull request merged**